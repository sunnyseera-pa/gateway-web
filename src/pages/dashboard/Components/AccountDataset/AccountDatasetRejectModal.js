import React from 'react';
import { Modal } from 'react-bootstrap';
import { ReactComponent as CloseButtonSvg } from '../../../../images/close-alt.svg';
import './AccountDatasetDecisionModal.scss';
import _ from 'lodash';

const AccountDatasetRejectModal = ({
	open,
	closed,
    onReject,
	onRejectAndGoToNext
}) => {
	return (
		<Modal
			show={open}
			onHide={closed}
			className='account-dataset-decision-modal'
			size='lg'
			aria-labelledby='contained-modal-title-vcenter'
			centered>
			<div className='account-dataset-decision-header'>
				<CloseButtonSvg className='account-dataset-decision-modal-close' onClick={closed} />
				<div className='account-dataset-decision-header--wrap'>
					<h4 className='black-20'>Reject this version of this dataset metadata</h4>
				</div>
			</div>

			<div className='account-dataset-decision-body'>
				<div className='account-dataset-decision-body--wrap'>Placeholder content for now</div>
			</div>
			<div className='account-dataset-decision-footer'>
				<div className='account-dataset-decision-footer--wrap'>
					<button
						className='button-secondary'
						style={{ marginLeft: 'auto' }}
						onClick={() => {
							closed();
						}}>
						No, nevermind
					</button>
					<button
						className='button-secondary'
						style={{ marginLeft: 'auto' }}
						onClick={() => {
							onReject();
							closed();
						}}>
						Reject
					</button>
					<button
						className='button-secondary'
						style={{ marginLeft: 'auto' }}
						onClick={() => {
							onRejectAndGoToNext();
							closed();
						}}>
						Reject and go to next
					</button>
				</div>
			</div>
		</Modal>
	);
};

export default AccountDatasetRejectModal;
