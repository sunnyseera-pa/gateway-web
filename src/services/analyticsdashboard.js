import { useQuery } from 'react-query';
import { apiURL } from '../configs/url.config';
import { getRequest } from '../utils/requests';

const getTotalUsers = options => {
	return getRequest(`${apiURL}/analyticsdashboard/totalusers`, options);
};

const getUsersPerMonth = options => {
	return getRequest(`${apiURL}/analyticsdashboard/userspermonth`, options);
};

const useGetTotalUsers = (requestOptions, queryOptions = { queryKey: 'getTotalUsers' }) => {
	return useQuery({
		...queryOptions,
		queryFn: () => getTotalUsers(requestOptions),
	});
};

const useGetUsersPerMonth = (requestOptions, queryOptions = { queryKey: 'getUsersPerMonth' }) => {
	return useQuery({
		...queryOptions,
		queryFn: () => getUsersPerMonth(requestOptions),
	});
};

export default {
	getTotalUsers,
	getUsersPerMonth,
	useGetTotalUsers,
	useGetUsersPerMonth,
};
