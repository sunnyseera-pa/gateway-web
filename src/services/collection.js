import { useMutation, useQuery } from 'react-query';
import { apiURL } from '../configs/url.config';
import { getRequest, postRequest } from '../utils/requests';

export const getCollectionRequest = (_id, options) => {
	return getRequest(`${apiURL}/collections/${_id}`, options);
};

export const getCollectionRelatedObjectsRequest = (_id, options) => {
	return getRequest(`${apiURL}/collections/relatedobjects/${_id}`, options);
};

export const postCollectionCounterUpdateRequest = (data, options) => {
	return postRequest(`${apiURL}/collectioncounter/update`, data, options);
};

const useGetCollectionRequest = (_id, requestOptions, queryOptions = { queryKey: 'getCollectionRequest' }) => {
	return useQuery({
		...queryOptions,
		queryKey: [queryOptions.queryKey, _id],
		queryFn: async ({ queryKey }) => getCollectionRequest(queryKey[1], requestOptions),
	});
};

const useGetCollectionRelatedObjectsRequest = (_id, requestOptions, queryOptions = { queryKey: 'getCollectionRelatedObjectsRequest' }) => {
	return useQuery({
		...queryOptions,
		queryKey: [queryOptions.queryKey, _id],
		queryFn: async ({ queryKey }) => getCollectionRelatedObjectsRequest(queryKey[1], requestOptions),
	});
};

const usePostCollectionCounterUpdateRequest = (requestOptions, mutateOptions = { queryKey: 'postCollectionCounterUpdateRequest' }) => {
	return useMutation(data => postCollectionCounterUpdateRequest(data, requestOptions), {
		mutateOptions,
	});
};

export default {
	getCollectionRequest,
	getCollectionRelatedObjectsRequest,
	postCollectionCounterUpdateRequest,
	useGetCollectionRequest,
	useGetCollectionRelatedObjectsRequest,
	usePostCollectionCounterUpdateRequest,
};
