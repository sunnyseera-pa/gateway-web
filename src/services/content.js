import { useQuery } from 'react-query';
import { getRequest } from '../utils/requests';
import { UATCMSURL } from './content.constants';

const baseURL = require('../pages/commonComponents/BaseURL');
const cmsURL = baseURL.getCMSURL();
const env = baseURL.getURLEnv();
const local = 'local';

export const getNon5SafesModalContentRequest = (options) => {
    const url = env === local ? UATCMSURL : cmsURL;
	return getRequest(`${url}/Non5SafesModalContent`, options);
};

const useGetNon5SafesModalContentRequest = (requestOptions, queryOptions = { queryKey: 'getNon5SafesModalContentRequest' }) => {
	return useQuery({
		...queryOptions,
		queryFn: () => getNon5SafesModalContentRequest(requestOptions),
	});
};

export default {
    useGetNon5SafesModalContentRequest
}
