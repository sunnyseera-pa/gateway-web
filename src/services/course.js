import { useMutation, useQuery } from 'react-query';
import { apiURL } from '../configs/url.config';
import { getRequest, postRequest, putRequest } from '../utils/requests';

const getCourse = (_id, options) => {
	return getRequest(`${apiURL}/course/${_id}`, options);
};

const getEdit = (_id, options) => {
	return getRequest(`${apiURL}/course/edit/${_id}`, options);
};

const postCourse = (data, options) => {
	return postRequest(`${apiURL}/course`, data, options);
};

const putCourse = (_id, data, options) => {
	return putRequest(`${apiURL}/course/${_id}`, data, options);
};

const useGetCourse = (requestOptions, queryOptions = { queryKey: 'getCourse' }) => {
	return useQuery({
		...queryOptions,
		queryFn: _id => getCourse(_id, requestOptions),
	});
};

const useGetEdit = (requestOptions, queryOptions = { queryKey: 'getEdit' }) => {
	return useQuery({
		...queryOptions,
		queryFn: _id => getEdit(_id, requestOptions),
	});
};

const usePostCourse = (requestOptions, mutateOptions = { queryKey: 'postCourse' }) => {
	return useMutation(data => postCourse(data, requestOptions), {
		mutateOptions,
	});
};

const usePutCourse = (requestOptions, mutateOptions = { queryKey: 'putCourse' }) => {
	return useMutation((_id, data) => putCourse(_id, data, requestOptions), {
		mutateOptions,
	});
};

export default {
	getCourse,
	getEdit,
	postCourse,
	putCourse,
	useGetCourse,
	useGetEdit,
	usePostCourse,
	usePutCourse,
};
