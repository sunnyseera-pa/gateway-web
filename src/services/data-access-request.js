import { useMutation, useQuery } from 'react-query';
import { apiURL } from '../configs/url.config';
import { deleteRequest, getRequest, patchRequest, postRequest, putRequest } from '../utils/requests';

const getDataAccessRequests = options => {
	return getRequest(`${apiURL}/data-access-request`, options);
};

const getDataAccessRequest = (_id, options) => {
	return getRequest(`${apiURL}/data-access-request/${_id}`, options);
};

const postDataAccessRequest = (_id, data, options) => {
	return postRequest(`${apiURL}/data-access-request/${_id}`, data, options);
};

const putDataAccessRequest = (_id, data, options) => {
	return putRequest(`${apiURL}/data-access-request/${_id}`, data, options);
};

const patchDataAccessRequest = (_id, data, options) => {
	return patchRequest(`${apiURL}/data-access-request/${_id}`, data, options);
};

const deleteDataAccessRequest = (_id, options) => {
	return deleteRequest(`${apiURL}/data-access-request/${_id}`, options);
};

const useGetDataAccessRequests = (requestOptions, queryOptions = { queryKey: 'getDataAccessRequest' }) => {
	return useQuery({
		...queryOptions,
		queryFn: () => getDataAccessRequests(requestOptions),
	});
};

const useGetDataAccessRequest = (requestOptions, queryOptions = { queryKey: 'getDataAccessRequest' }) => {
	return useQuery({
		...queryOptions,
		queryFn: id => getDataAccessRequest(id, requestOptions),
	});
};

const usePostDataAccessRequest = (requestOptions, mutateOptions = { queryKey: 'postDataAccessRequest' }) => {
	return useMutation((id, data) => postDataAccessRequest(id, data, requestOptions), {
		mutateOptions,
	});
};

const usePutDataAccessRequest = (requestOptions, mutateOptions = { queryKey: 'putDataAccessRequest' }) => {
	return useMutation((id, data) => putDataAccessRequest(id, data, requestOptions), {
		mutateOptions,
	});
};

const usePatchDataAccessRequest = (requestOptions, mutateOptions = { queryKey: 'patchDataAccessRequest' }) => {
	return useMutation((id, data) => patchDataAccessRequest(id, data, requestOptions), {
		mutateOptions,
	});
};

const useDeleteDataAccessRequest = (requestOptions, queryOptions = { queryKey: 'deleteDataAccessRequest' }) => {
	return useQuery({
		...queryOptions,
		queryFn: id => deleteDataAccessRequest(id, requestOptions),
	});
};

export default {
	getDataAccessRequests,
	getDataAccessRequest,
	postDataAccessRequest,
	putDataAccessRequest,
	patchDataAccessRequest,
	deleteDataAccessRequest,
	useGetDataAccessRequests,
	useGetDataAccessRequests,
	useGetDataAccessRequest,
	usePostDataAccessRequest,
	usePutDataAccessRequest,
	usePatchDataAccessRequest,
	useDeleteDataAccessRequest,
};
