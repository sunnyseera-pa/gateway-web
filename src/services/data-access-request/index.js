export { default } from './data-access-request';
