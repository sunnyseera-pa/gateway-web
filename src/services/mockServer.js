import { setupServer } from 'msw/node';
import { rest } from 'msw';
import mswDatasets from './datasets/mockMsw';
import mswDatasetOnboarding from './dataset-onboarding/mockMsw';
import mswPostDatasetActivityLog from './activitylog/mockMsw';
import mswSearch from './search/mockMsw';
import mswGetLocations from './locations/mockMsw';
import mswGetContributors from './contributors/mockMsw';

import translations from '../../public/locales/en-GB/translation.json';

const mswGetEnTranslations = rest.get(`http://localhost/locales/en/translation.json`, (req, res, ctx) => {
	return res(ctx.status(200), ctx.json(translations));
});

const mswGetEnGbTranslations = rest.get(`http://localhost/locales/en-GB/translation.json`, (req, res, ctx) => {
	return res(ctx.status(200), ctx.json(translations));
});

const mswGetIcon = rest.get(`/images/Application_approved.svg`, (req, res, ctx) => {
	return res(ctx.status(200), ctx.text('<svg />'));
});

const handlers = [
	...mswDatasets,
	...mswDatasetOnboarding,
	...mswPostDatasetActivityLog,
	// ...mswSearch,
	...mswGetLocations,
	...mswGetContributors,
	mswGetEnTranslations,
	mswGetEnGbTranslations,
	mswGetIcon,
];

export const server = setupServer(...handlers);
