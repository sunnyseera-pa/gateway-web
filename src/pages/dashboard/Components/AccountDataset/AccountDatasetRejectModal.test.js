import { render, waitFor, within } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import userEvent from "@testing-library/user-event";
import React from 'react';
import AccountDatasetRejectModal from './AccountDatasetRejectModal';
import datasetOnboardingService from '../../../../services/dataset-onboarding/dataset-onboarding';
import { server } from '../../../../services/mockServer';

jest.mock('../../../../services/dataset-onboarding/dataset-onboarding');

let containerDiv;
const goToNext = jest.fn();
const closed = jest.fn();

describe('Given the AccountDatasetRejectModal component', () => {

    const props = {
        id: "id",
        open: true,
        closed,
        goToNext,
        showGoToNext: true
    };

    describe('When it is rendered', () => {
        let wrapper;

        beforeAll(() => {
            server.listen();
            containerDiv = createPortalContainer();
            wrapper = render(
                <AccountDatasetRejectModal {...props} container={containerDiv} />, {
                    wrapper: Providers,
                }
            );
        });

        afterEach(() => {
            server.resetHandlers();
        });

        afterAll(() => {
            server.close();
			removePortalContainer(containerDiv);
		});

        it('Should match the snapshot', async () => {
            expect(containerDiv).toMatchSnapshot();
        });

        it('Then the Reject and go to next button should not be disabled', async () => {
            await waitFor(() => expect(wrapper.getByText('Reject and go to next')).toBeTruthy());

            const { getByText } = wrapper;
            const rejectAndGoToNextButton = getByText('Reject and go to next');
            expect(rejectAndGoToNextButton).not.toHaveAttribute('disabled');
        });

        describe('And showGoToNext is false', () => {
            beforeAll(() => {
                const { rerender } = wrapper;
                const newProps = {
                    ...props,
                    showGoToNext: false
                };

                rerender(
                    <AccountDatasetRejectModal {...newProps} container={containerDiv} />, {
                        wrapper: Providers,
                    }
                )
            });
            it('Then the Reject and go to next button should be disabled', async () => {
                await waitFor(() => expect(wrapper.getByText('Reject and go to next')).toBeTruthy());

                const { getByText } = wrapper;
                const rejectAndGoToNextButton = getByText('Reject and go to next');
                expect(rejectAndGoToNextButton).toHaveAttribute('disabled');
            });
        });

        describe('And the Reject button is clicked', () => {
            let button;

            beforeAll(async () => {
                const { rerender } = wrapper;

                rerender(
                    <AccountDatasetRejectModal {...props} container={containerDiv} />, {
                        wrapper: Providers,
                    }
                )

                await waitFor(() => expect(wrapper.getByText('Reject and go to next')).toBeTruthy());

                const { getByTestId, getByRole } = wrapper;

                const descriptionTextArea = getByRole('textarea');
                userEvent.type(descriptionTextArea, "rejected");

                button = within(getByTestId('button-container')).getAllByText('Reject')[0];
                userEvent.click(button);
            });

            it('Then submits the dataset rejection request', async () => {
                await waitFor(() => expect(datasetOnboardingService.usePutDatasetOnboarding).toHaveBeenCalledWith({
                    id: 'id',
                    applicationStatus: 'rejected',
                    applicationStatusDesc: ''
                }));
            });

            it('Then closes the modal', async () => {
                await waitFor(() => expect(closed).toHaveBeenCalled());
            });
        });

        describe('And the Reject and go to next button is clicked', () => {
            let button;

            beforeAll(async () => {
                const { rerender } = wrapper;

                rerender(
                    <AccountDatasetRejectModal {...props} container={containerDiv} />, {
                        wrapper: Providers,
                    }
                )

                await waitFor(() => expect(wrapper.getByText('Reject and go to next')).toBeTruthy());
                
                const { getByText, getByTestId } = wrapper;

                const descriptionTextArea = getByTestId('dataset-reject-applicationStatusDesc');
                fireEvent.change(descriptionTextArea, { target: { value: 'rejected' } });

                button = getByText('Reject and go to next');
                fireEvent.click(button);
            });

            it('Then submits the dataset rejection request', async () => {
                await waitFor(() => expect(datasetOnboardingService.usePutDatasetOnboarding).toHaveBeenCalledWith({
                    id: 'id',
                    applicationStatus: 'rejected',
                    applicationStatusDesc: ''
                }));
            });

            it('Then goes to next dataset', async () => {
                await waitFor(() => expect(goToNext).toHaveBeenCalled());
            });

            it('Then closes the modal', async () => {
                await waitFor(() => expect(closed).toHaveBeenCalled());
            });
        });
    });
});
