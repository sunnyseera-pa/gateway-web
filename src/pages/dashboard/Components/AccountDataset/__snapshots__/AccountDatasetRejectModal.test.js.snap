// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given the AccountDatasetRejectModal component When it is rendered Should match the snapshot 1`] = `
<div
  aria-hidden="true"
/>
`;
